Basic check to make sure the bulk import seperator is being used.

tonypartridge opened the pull request:
Added Import State function as to how the urls should be imported, enabled or disabled.

Pull Request for Issue #18844.

### Summary of Changes

Bulk Importer now has a separator missing warning when bulk import occurs without the seperator anywhere within the string.

Added Bulk Import Enable / Disable links on import by default.

Commit:
Basic check to make sure the bulk import seperator is being used.
Added Import State function as to how the urls should be imported, enabled or disabled.

Commit:
force int.

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,15 @@ public function batch()
 			{
 				$params = JComponentHelper::getParams('com_redirect');
-				$batch_urls[] = array_map('trim', explode($params->get('separator', '|'), $batch_urls_line));
+				$seperator  = $params->get('separator', '|');
```
Spelling `$separator`

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,15 @@ public function batch()
+				$seperator  = $params->get('separator', '|');
+
+				// Basic check to make sure the correct separator is being used
+				if (!\Joomla\String\StringHelper::strpos($batch_urls_line, $seperator)) {
```
Move `{` to its own line.

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,15 @@ public function batch()
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					return false;
+				}
+				$batch_urls[] = array_map('trim', explode($seperator, $batch_urls_line));
```
Add blank line above.

Quy commented on administrator/language/en-GB/en-GB.com_redirect.ini:
```diff
@@ -84,6 +86,7 @@ COM_REDIRECT_N_LINKS_UPDATED_1="1 link has been updated."
 COM_REDIRECT_NEW_LINK="New Link"
 COM_REDIRECT_NO_ITEM_ADDED="No links added."
 COM_REDIRECT_NO_ITEM_SELECTED="No links selected."
+COM_REDIRECT_NO_SEPERATOR_FOUND="The seperator %s was not found in your import."
```
Spelling `separator`

Quy commented on administrator/components/com_redirect/config.xml:
```diff
@@ -22,6 +22,17 @@
 			default="|"
 		/>
+		<field
+				name="defaultImportState"
```
Fix whitespace

tonypartridge committed:
Update config.xml

tonypartridge committed:
Update links.php

tonypartridge committed:
Update en-GB.com_redirect.ini

tonypartridge commented:
Thanks @Quy clearly tired lastnight :/

Quy commented on administrator/components/com_redirect/config.xml:
```diff
@@ -22,6 +22,17 @@
+			description="COM_REDIRECT_DEFAULT_IMPORT_STATE_DESC"
+			class="btn-group btn-group-yesno"
+			default="0"
+		>
```
Indent

tonypartridge replied:
How do you mean? the > finishes inline with the opening <?

Quy replied:
It should be line 14 before your change. Check the other XML files to see the format when there are `<option>`.

tonypartridge replied:
Thanks! That's interesting I always thought they were double indented. I must get PHPCS setup on my MBP. Thanks @Quy what would we do without you! 

Are you a real person or someones AI Bot? ;-)

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,16 @@ public function batch()
+				{
+					$this->setMessage(JText::sprintf('COM_REDIRECT_NO_SEPARATOR_FOUND', $separator), 'error');
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					return false;
```
Add blank line above.

tonypartridge replied:
tnanks

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,16 @@ public function batch()
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					return false;
+				}
+				$batch_urls[] = array_map('trim', explode($separator, $batch_urls_line));
```
Add blank line above.

tonypartridge replied:
doh thanks.

Quy commented on administrator/language/en-GB/en-GB.com_redirect.ini:
```diff
@@ -84,6 +86,7 @@ COM_REDIRECT_N_LINKS_UPDATED_1="1 link has been updated."
 COM_REDIRECT_NEW_LINK="New Link"
 COM_REDIRECT_NO_ITEM_ADDED="No links added."
 COM_REDIRECT_NO_ITEM_SELECTED="No links selected."
+COM_REDIRECT_NO_SEPARATOR_FOUND="The seperator %s was not found in your import."
```
Fix `seperator`

tonypartridge replied:
Fixed thanks

tonypartridge committed:
Update config.xml

tonypartridge committed:
Update links.php

tonypartridge committed:
Update en-GB.com_redirect.ini

tonypartridge committed:
Update config.xml

As per standards i.e:
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_config/model/form/application.xml

i.e. endtag inline with options and closing tag inline with opening tag.

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,18 @@ public function batch()
+				{
+					$this->setMessage(JText::sprintf('COM_REDIRECT_NO_SEPARATOR_FOUND', $separator), 'error');
+					$this->setRedirect('index.php?option=com_redirect&view=links');
+					
```
Remove tabs.

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,18 @@ public function batch()
+					
+					return false;
+				}
+				
```
Remove tabs.

Quy commented on administrator/components/com_redirect/controllers/links.php:
```diff
@@ -133,7 +133,18 @@ public function batch()
 			{
 				$params = JComponentHelper::getParams('com_redirect');
-				$batch_urls[] = array_map('trim', explode($params->get('separator', '|'), $batch_urls_line));
+				$separator  = $params->get('separator', '|');
```
Align or remove the extra space before `=`.

tonypartridge committed:
Update links.php

tonypartridge commented:
@Quy done thanks!

Quy commented:
I have tested this item :white_check_mark: successfully on b2cd2e183f1a58c18d3777251ad3d9514194c7be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19982">issues.joomla.org/tracker/joomla-cms/19982</a>.</sub>

anuragteapot commented:
I have tested this item :white_check_mark: successfully on b2cd2e183f1a58c18d3777251ad3d9514194c7be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19982">issues.joomla.org/tracker/joomla-cms/19982</a>.</sub>

ghost commented:
Ready to Commit after two successful tests.
